build/ci/nightly/Jenkinsfile

```diff
@@ -0,0 +1,59 @@
+pipeline {
+
+    agent {
+        label 'linux'
+    }
+
+    options {
+        timeout(time: 1, unit: 'HOURS')
+    }
+
+    environment {
+        VAULT_ADDR = credentials('vault-addr')
+        VAULT_ROLE_ID = credentials('vault-role-id')
+        VAULT_SECRET_ID = credentials('vault-secret-id')
+        GCLOUD_PROJECT = credentials('k8s-operators-gcloud-project')
+        REGISTRY = "push.docker.elastic.co"
+        REPOSITORY = "eck-snapshots"
+        IMG_NAME = "eck-operator"
+        SNAPSHOT = "true"
+        DOCKER_IMAGE_NO_TAG = "docker.elastic.co/${REPOSITORY}/${IMG_NAME}"
+    }
+
+    stages {
+        stage('Run unit and integration tests') {
+            steps {
+                sh 'make -C build/ci ci-pr'
+            }
+        }
+        stage('Build and push Docker image') {
+            steps {
+                sh """
+                    export VERSION=\$(cat $WORKSPACE/operators/VERSION)-\$(date +%F)-\$(git rev-parse --short --verify HEAD)
+                    export OPERATOR_IMAGE=${REGISTRY}/${REPOSITORY}/${IMG_NAME}:\$VERSION
+                    make -C build/ci ci-release
+                """
+            }
+        }
+    }
+
+    post {
+        success {
+            script {
+                def version = sh(returnStdout: true, script: 'cat $WORKSPACE/operators/VERSION')
+                def hash = sh(returnStdout: true, script: 'git rev-parse --short --verify HEAD')
+                def date = new Date()
+                def image = env.DOCKER_IMAGE_NO_TAG + ":" + version + "-" + date.format("yyyy-MM-dd") + "-" + hash
+                currentBuild.description = image
```

Comment:
Should this be there for all the builds (not only the successful ones)?

Reply:
This part is just adding Docker image name to job description for easy discoverability. If job failed, then there will be no Docker image and there is no need, in my opinion, to add a name of unexisted Docker image to job description

Reply:
I was thinking that this will allow us to map failed build to a particular hash easier, but I'm fine with this as it too.